src/platform/packages/shared/kbn-esql-ast/src/mutate/commands/rerank/index.ts

```
@@ -53,19 +57,88 @@ export const setFields = (
+  const toExpression = (
+    val: string | ESQLLiteral | ESQLParamLiteral
+  ) => {
+    if (typeof val === 'string') {
+      return val.startsWith('?')
+        ? Builder.param.build(val)
+        : Builder.expression.literal.string(val);
+    }
+    return val;
+  };
+
 
-  if (id.type !== 'identifier' && id.type !== 'literal') {
-    throw new Error(`Invalid RERANK inferenceId: ${id}`);
+  // Type guard to check if an AST item is a WITH option
+  const isWithOption = (arg: ESQLAstItem): arg is ESQLCommandOption =>
+    !!arg && !Array.isArray(arg) && arg.type === 'option' && arg.name === 'with';
+
+  // Validates and retrieves the map from a WITH option
+  const getWithOptionMap = (withOption: ESQLCommandOption): ESQLMap => {
+    const mapArg = withOption.args[0];
+
+    if (!mapArg || typeof mapArg === 'string' || Array.isArray(mapArg) || mapArg.type !== 'map') {
+      throw new Error('WITH option must contain a map');
+    }
+
+    return mapArg as ESQLMap;
+  };
+
+  // Normalizes a key for comparison by removing quotes
+  const normalizeKey = (keyValue: string | undefined): string => {
+    return keyValue?.replace(/"/g, '') ?? '';
+  };
+
+  // Checks if an entry in the map has the specified key
+  // normalizeKey avoid cases like: '"inference_id" === "inferenceId"' -> this is false
+  const getExistingEntry = (entry: ESQLMapEntry) => normalizeKey(entry.key.valueUnquoted ?? entry.key.value) === key;
+
+  // Creates a new map entry with the given key and value
+  const createMapEntry = (key: string, value: ESQLLiteral | ESQLParamLiteral) => ({
+    type: 'map-entry' as const,
+    name: 'map-entry' as const,
+    key: Builder.expression.literal.string(key),
+    value,
+    location: { min: 0, max: 0 },
+    text: '',
+    incomplete: false,
+  });
+
+  // Find the WITH option in the command arguments
+  const withOption = cmd.args.find(isWithOption);
+
+  if (!withOption) {
+    throw new Error('RERANK command must have a WITH option');
```

I haven't seen any other cases using  throw new Error, but I think failing silently isn't a good idea.